test/conformance/source/environment.cpp

```diff
@@ -199,6 +201,14 @@ DevicesEnvironment::DevicesEnvironment(int argc, char **argv)
         error = "Could not find any devices associated with the platform";
         return;
     }
+    // Get an environment variable (NUMBER_OF_DEVICES) to run tests on the appropriate number of devices
+    auto devices_num = ur_getenv("NUMBER_OF_DEVICES");
+    int count_set = devices_num ? std::stoi(devices_num.value()) : 1;
```

Comment:
we should catch the exception(s) here - it's an input from env var, after all

Reply:
I have changed it.